app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list:
+    person_list = [
+        Person.people.get(
+            person.get("name"),
+            Person(person.get("name"), person.get("age"))
+        ) for person in people
+    ]
+
+    for person in people:
+        if "wife" in person:
+            name_of_wife_or_husband = person["wife"]
+        else:
+            name_of_wife_or_husband = person["husband"]
```

Comment:
Remove this and simplify by just two conditions:
1. if person.get("wife") is not None
2. if person.get("husband") is not None